shared/js/l10n.js

```diff
@@ -1374,11 +1390,14 @@
   Context.prototype.registerLocales = function(defLocale, available) {
     /* jshint boss:true */
     this.availableLocales = [this.defaultLocale = defLocale];
+    this.localeSources[defLocale] =
+      (available && (defLocale in available)) ? available[defLocale] : 'bundle';
 
     if (available) {
-      for (var i = 0, loc; loc = available[i]; i++) {
+      for (var loc in available) {
```

Comment:
I recall at least one case when `Object.prototype` was extended in the Cost Control app, which at the time broke our code like this.  I think it's safer to use the `var i = 0` idiom for looping here, even if I'm not a fan of it at all.